azurerm/resource_arm_container_group.go

```diff
@@ -1241,23 +1241,26 @@ func expandContainerGroupDiagnostics(input []interface{}) *containerinstance.Con
 
 	workspaceId := analyticsV["workspace_id"].(string)
 	workspaceKey := analyticsV["workspace_key"].(string)
-	logType := containerinstance.LogAnalyticsLogType(analyticsV["log_type"].(string))
 
-	metadataMap := analyticsV["metadata"].(map[string]interface{})
-	metadata := make(map[string]*string)
-	for k, v := range metadataMap {
-		strValue := v.(string)
-		metadata[k] = &strValue
+	logAnalytics := containerinstance.LogAnalytics{
+		WorkspaceID:  utils.String(workspaceId),
+		WorkspaceKey: utils.String(workspaceKey),
 	}
 
-	return &containerinstance.ContainerGroupDiagnostics{
-		LogAnalytics: &containerinstance.LogAnalytics{
-			WorkspaceID:  utils.String(workspaceId),
-			WorkspaceKey: utils.String(workspaceKey),
-			LogType:      logType,
-			Metadata:     metadata,
-		},
+	if logType := analyticsV["log_type"].(string); logType != "" {
+		logAnalytics.LogType = containerinstance.LogAnalyticsLogType(logType)
+
+		metadataMap := analyticsV["metadata"].(map[string]interface{})
+		metadata := make(map[string]*string)
+		for k, v := range metadataMap {
+			strValue := v.(string)
+			metadata[k] = &strValue
+		}
+
+		logAnalytics.Metadata = metadata
```

Comment:
double-checking this: is it possible to set the Metadata without a log type?

Reply:
No – if you try to do it, Azure responds with status 400 and a message saying that you can't set metadata with log type `'<null>'`. I'll look into creating an acceptance test.